app/controllers/api/connect/v3/systems/products_controller.rb

```diff
@@ -124,7 +124,7 @@ def remove_previous_product_activations(product_ids)
 
   def load_subscription
     # Find subscription by regcode if provided, otherwise use the first subscription (bsc#1220109)
```

Comment:
```suggestion
        # Find subscription by regcode if provided and not a public cloud system, 
        # otherwise check if there's already an activation with a matching 
        # subscription (for migrations, bsc#1220109)
```